tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -155,44 +136,146 @@ var _ = Describe("odo devfile push command tests", func() {
 			var statErr error
 			oc.CheckCmdOpInRemoteDevfilePod(
 				podName,
+				"",
 				namespace,
-				[]string{"stat", "/projects/server.js"},
+				[]string{"stat", "/projects/nodejs-web-app/app/app.js"},
 				func(cmdOp string, err error) bool {
 					statErr = err
 					return true
 				},
 			)
 			Expect(statErr).ToNot(HaveOccurred())
-			Expect(os.Remove(filepath.Join(context, "server.js"))).NotTo(HaveOccurred())
+			Expect(os.Remove(filepath.Join(context, "app", "app.js"))).NotTo(HaveOccurred())
 			helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--namespace", namespace)
 
 			oc.CheckCmdOpInRemoteDevfilePod(
 				podName,
+				"",
 				namespace,
-				[]string{"stat", "/projects/server.js"},
+				[]string{"stat", "/projects/nodejs-web-app/app/app.js"},
 				func(cmdOp string, err error) bool {
 					statErr = err
 					return true
 				},
 			)
 			Expect(statErr).To(HaveOccurred())
-			Expect(statErr.Error()).To(ContainSubstring("cannot stat '/projects/server.js': No such file or directory"))
+			Expect(statErr.Error()).To(ContainSubstring("cannot stat '/projects/nodejs-web-app/app/app.js': No such file or directory"))
 		})
 
 		It("should build when no changes are detected in the directory and force flag is enabled", func() {
-			// Devfile push requires experimental mode to be set
-			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
 
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs-multicontainer"), context)
-			cmpName := helper.RandString(6)
-			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, "--project", namespace, cmpName)
 			helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--namespace", namespace)
 
 			// use the force build flag and push
 			output := helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--namespace", namespace, "-f")
 			Expect(output).To(Not(ContainSubstring("No file changes detected, skipping build")))
 		})
 
+		It("should execute the default devbuild and devrun commands if present", func() {
```

Comment:
Can some of these test cases test that the built/running application is actually running? 

We might need to ask the odo folks (tag @amitkrout) if it's possible to get the ingress domain and pass it into `odo url create`, if not we can do one of three things:
1) Wait for `odo url create` to support creating routes for devfile components on OpenShift (so we don't have to pass in the ingress domain)
2) Exec into the runtime container and do a curl/wget there
3) Handle later as part of the end-to-end tests

Reply:
Thinking aloud, how do we configure ingress host between running on CI and running on ppl's machines?

Reply:
@maysunfaisal It looks like https://github.com/openshift/odo/issues/2833 is the issue we wanna keep an eye on. So maybe best to wait?

Reply:
yeah url endpoint testing can be done with that issue!